src/main/java/com/serotonin/bacnet4j/test/DaqTest/VersionTest.java

```diff
@@ -0,0 +1,118 @@
+package com.serotonin.bacnet4j.test.DaqTest;
+
+import java.util.Map;
+import com.serotonin.bacnet4j.LocalDevice;
+import com.serotonin.bacnet4j.RemoteDevice;
+import com.serotonin.bacnet4j.exception.BACnetException;
+import com.serotonin.bacnet4j.npdu.ip.IpNetwork;
+import com.serotonin.bacnet4j.service.unconfirmed.WhoIsRequest;
+import com.serotonin.bacnet4j.test.DaqTest.helper.Connection;
+import com.serotonin.bacnet4j.test.DaqTest.helper.Report;
+import com.serotonin.bacnet4j.type.Encodable;
+import com.serotonin.bacnet4j.type.enumerated.PropertyIdentifier;
+import com.serotonin.bacnet4j.util.RequestUtils;
+
+public class VersionTest {
+
+  private Connection connection;
+  private static LocalDevice localDevice;
+  private String localIp = "";
+  private String broadcastIp = "";
+
+  private String appendixText = "";
+  private boolean testPassed = false;
+  private String passedReportText = "RESULT pass protocol.bacnet.version";
+  private String failedReportText = "RESULT fail protocol.bacnet.version";
+  private String errorPropertyMessage = "errorClass=Property, errorCode=Unknown property";
+
+  public VersionTest(String localIp, String broadcastIp) throws Exception {
+    this.localIp = localIp;
+    this.broadcastIp = broadcastIp;
+    discoverDevices();
+  }
+
+  public void discoverDevices() throws Exception {
+    connection = new Connection(broadcastIp, IpNetwork.DEFAULT_PORT, localIp);
+    while (!connection.isTerminate()) {
+      localDevice = connection.getLocalDevice();
+      System.err.println("Sending whois...");
+      localDevice.sendGlobalBroadcast(new WhoIsRequest());
+      // Wait a bit for responses to come in.
+      System.err.println("Waiting...");
+      Thread.sleep(5000);
+      System.err.println("Processing...");
+      getVersion();
+      connection.doTerminate();
+    }
+  }
+
+  private void getVersion() {
```

Comment:
function is doing more than "getVersion" -- since it's also writing the report.  Try to name functions appropriately for what they do.